examples/index.ts

```diff
@@ -39,9 +39,9 @@ const main = async () => {
   }
 
   console.log('Listing caches:');
-  let token;
+  let token: string | undefined;
   do {
-    const listResponse = await momento.listCaches();
+    const listResponse = await momento.listCaches(token);
```

Comment:
lol wow nice catch, thanks!

Reply:
oh wait, so what happens the first time through the loop here?  We are passing token in as `undefined`, but that works and we still get the list successfully?

Reply:
It works the same as if the variable hadn't been passed in as per example 4 here:
[https://stackoverflow.com/questions/57243477/can-an-optional-parameter-be-null-in-typescript](url)

It does look like the runtime can be configured to disallow this with `exactOptionalPropertyTypes`:
[https://www.typescriptlang.org/tsconfig#exactOptionalPropertyTypes](url)

The example works in my testing, but I don't know what is idiomatic here.

Reply:
That line of code won't do anything without pagination working. The actual infinte loop that is fixed here is on line 54. This is just future proofing.